agent/util-scripts/pbench-register-tool-set

```diff
@@ -129,83 +129,12 @@ if [[ -z ${the_tool_set} ]]; then
 	exit 1
 fi
 
-
-# This function checks a --remotes file argument  and if everything
-# goes OK it returns 0 (it also returns 0 if the --remotes argument is
-# not a file).  Otherwise, it logs an error and returns 1.
-
-# This is an emasculated version of the code that pbench-register-tool
-# uses to do this checking: in addition to the checking, that code
-# gathers information from the file into a couple of arrays. Since we
-# don't need those for pbench-register-tool-set the data gathering is
-# elided in this function.
-
-function check_remotes_file () {
-	local remotes_arg="${1}"
-	local lineno
-
-	if [[ ${remotes_arg::1} == "@" ]]; then
-		# We are dealing with a file containing the list of remotes.
-
-		if [[ -n ${labels_arg} ]]; then
-			error_log "--labels=${labels_arg} not allowed with remotes file (${remotes_arg})"
-			return 1
-		fi
-		remotes_file=${remotes_arg#*@}
-		if [[ ! -e ${remotes_file} ]]; then
-			error_log "--remotes=@${remotes_file} specifies a file that does not exist"
-			return 1
-		fi
-
-		lineno=0
-		while read line; do
-			(( lineno++ ))
-			if [[ ${line::1} == "#" ]]; then
-				# Ignore comments
-				(( lineno++ ))
-				continue
-			fi
-			IFS=',' read -ra parts <<< "${line}"
-			case "${#parts[@]}" in
-				0|1|2)
-				;;
-				*)
-				error_log "--remotes=@${remotes_file} contains an invalid file format, expected lines with \"<hostname>[,<label>]\" at line #${lineno}"
-				return 1
-				;;
-			esac
-		done < ${remotes_file}
-	fi
-
-	return 0
-}
-
-# call the function to check the argument
-check_remotes_file "${remotes_arg}"
-ret="${?}"
-if [[ "${ret}" != 0 ]]; then
+pbench-register-tool --name=vmstat --test-args ${remotes_arg}${labels_arg}
+if [[ ${?} -ne 0 ]]; then
 	usage >&2
-        exit 1
-fi
-
-# if we made it this far, everything is OK - prepare to pass the remotes (if specified) to pbench-register-tool 
-if [[ -n "${remotes_arg}" ]]; then
-	remotes_arg=" --remotes=$remotes_arg"
-fi
-
-if [[ -n "${labels_arg}" ]]; then
-	pbench-register-tool --name=perf --test-labels ${remotes_arg}${labels_arg}
-	if [[ ${?} -ne 0 ]]; then
-		if [[ -z "${remotes_arg}" ]]; then
-			remotes_arg="(default) --remotes=${_pbench_full_hostname}"
-		fi
-		error_log "The number of labels given, \"${labels_arg}\", does not match the number of remotes given, \"${remotes_arg}\""
-		usage >&2
-		exit 1
-	fi
+	exit 1
```

Comment:
It seems like it might be better to arrange for `pbench-register-tool` to print _its_ usage here, rather than suppressing that and printing  `pbench-register-tool-set`'s here...maybe?

Reply:
That would be strange: I invoke one program and I get the usage message from a different one? `pbench-register-tool` would issue its error messages when warranted, but ultimately I think the usage message should be the one from `pbench-register-tool-set`.

Reply:
The dissonance comes from the fact that `pbench-register-tool-set` is printing _its_ usage information because of a failure in `pbench-register-tool`...that is, we're hoping and presuming that `pbench-register-tool` will behave better if the user invokes `pbench-register-tool-set` differently...which is probably appropriate _almost_ all the time...but....

I'm thinking this might be a good time for `pbench-register-tool` to have a special, alternate error status which it uses to indicate failures specific to `--test-args`.  This would go a long way toward helping `pbench-register-tool-set` report the appropriate failure:  that is, if `pbench-register-tool` is invoked incorrectly (due to a bug in `pbench-register-tool-set`), it will return a `1` or a `2`; however, if `pbench-register-tool` wants to tell `pbench-register-tool-set` that the argument test failed, it should return a `3` (or equivalent).

Does that make sense?  (I'm not sure how this will actually play out in terms of the `pbench-register-tool` implementation, but it strikes me as a good idea in terms of its _interface_.)